### PR TITLE
Fix JNI emitSignal with one or more parameters (3.2)

### PR DESCRIPTION
**Pull request by timoschwarzer**
Cherry-pick of https://github.com/godotengine/godot/pull/39028 for 3.2

### CODE REVIEW

**Comment by timoschwarzer** [comment]
Related issue: https://github.com/godotengine/godot/pull/39028

File: platform/android/plugin/godot_plugin_jni.cpp [hunk 1]
```diff
@@ -119,17 +119,19 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_plugin_GodotPlugin_nativeEmitS
...
-	for (int i = 0; i < count; i++) {
+	ERR_FAIL_COND_MSG(count > VARIANT_ARG_MAX, "Maximum argument count exceeded!");
+
+	for (int i = 0; i < MIN(count, VARIANT_ARG_MAX); i++) {
 
 		jobject j_param = env->GetObjectArrayElement(j_signal_params, i);
-		Variant variant = _jobject_to_variant(env, j_param);
-		args[i] = &variant;
+		if (j_param)
+			args[i] = _jobject_to_variant(env, j_param);
```
Comment (pouleyKetchoupp) [inline platform/android/plugin/godot_plugin_jni.cpp:130]: Same thing with the missing brackets in this PR :)

Comment (timoschwarzer) [reply platform/android/plugin/godot_plugin_jni.cpp:130]: Ah, I thought the new code style only applies from 4.0 onwards, fixed :)

Comment (pouleyKetchoupp) [reply platform/android/plugin/godot_plugin_jni.cpp:130]: Thanks! And I guess it's true, I didn't think about that. In any case it's probably best to match the code between these 2 PRs as much as possible.

Comment (akien-mga) [reply platform/android/plugin/godot_plugin_jni.cpp:130]: The new code style is enforced in the `master` branch, but indeed there's no reason not to follow it when adding new code in `3.2` too.

Comment (timoschwarzer) [reply platform/android/plugin/godot_plugin_jni.cpp:130]: alright :)

**Comment by m4gr3d** [comment]
Replaced by #39047.

@timoschwarzer Thanks for highlighting the issue.
While this PR provides a fix to the issue, it unnecessarily limits the number of parameters that can be passed for a signal to `VARIANT_ARG_MAX`.

**Comment by timoschwarzer** [comment]
@m4gr3d thanks :) This should be changed in the JNI method for `GodotLib.calldeferred`, too, shouldn't it?

**Comment by m4gr3d** [comment]
> Edit: Just saw that `Object::_call_deferred_bind` is protected.

yep :) that prevents us from using a similar logic.

**Pull request closed without merging**